cognite/client/data_classes/transformations/common.py

```diff
@@ -305,6 +367,63 @@ def dump(self, camel_case: bool = False) -> Dict[str, Any]:
         """
         return basic_obj_dump(self, camel_case)
 
+class SourceOidcCredentials:
+    def __init__(
+        self,
+        client_id: str = None,
+        client_secret: str = None,
+        scopes: str = None,
+        token_uri: str = None,
+        audience: str = None,
+        cdf_project_name: str = None,
+    ):
+
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scopes = scopes
+        self.token_uri = token_uri
+        self.audience = audience
+        self.cdf_project_name = cdf_project_name
+
+    def dump(self, camel_case: bool = False) -> Dict[str, Any]:
+        """Dump the instance into a json serializable Python data type.
+
+        Args:
+            camel_case (bool): Use camelCase for attribute names. Defaults to False.
+
+        Returns:
+            Dict[str, Any]: A dictionary representation of the instance.
+        """
+        return basic_obj_dump(self, camel_case)
+
+class DestinationOidcCredentials:
+    def __init__(
+        self,
+        client_id: str = None,
+        client_secret: str = None,
+        scopes: str = None,
+        token_uri: str = None,
+        audience: str = None,
+        cdf_project_name: str = None,
+    ):
+
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scopes = scopes
+        self.token_uri = token_uri
+        self.audience = audience
+        self.cdf_project_name = cdf_project_name
+
+    def dump(self, camel_case: bool = False) -> Dict[str, Any]:
+        """Dump the instance into a json serializable Python data type.
+
+        Args:
+            camel_case (bool): Use camelCase for attribute names. Defaults to False.
+
+        Returns:
+            Dict[str, Any]: A dictionary representation of the instance.
+        """
+        return basic_obj_dump(self, camel_case)
```

Comment:
These classes are exactly the same as `OidcCredentials`, so they're redundant. Please remove them and use `OidcCredentials` instead.

Reply:
@erlendvollset Two classes `SourceOidcCredentials `and `DestinationOidcCredentials `defined because, client OIDC config override read and write authentication because of using `OidcCredentials `class . That was the issue. I will describe the issue and the fix beefily in a new PR, since this PR includes data centric destination and class renaming deployments.